mmv1/products/compute/InstanceGroupMembership.yaml

```diff
@@ -0,0 +1,106 @@
+read_verb: :POST
+self_link: projects/{{project}}/zones/{{zone}}/instanceGroups/{{instance_group}}/listInstances
+identity:
+  - instance
+nested_query: !ruby/object:Api::Resource::NestedQuery
+  keys:
+    - items
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Add instances': 'https://cloud.google.com/compute/docs/reference/rest/v1/instanceGroups/addInstances'
+    'Remove instances': 'https://cloud.google.com/compute/docs/reference/rest/v1/instanceGroups/removeInstances'
+    'List instances': 'https://cloud.google.com/compute/docs/reference/rest/v1/instanceGroups/listInstances'
+  api: 'https://cloud.google.com/compute/docs/reference/rest/v1/instanceGroups'
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    kind: 'compute#operation'
+    path: 'name'
+    base_url: 'projects/{{project}}/zones/{{zone}}/operations/{{op_id}}'
+    wait_ms: 1000
+  result: !ruby/object:Api::OpAsync::Result
+    path: 'targetLink'
+  status: !ruby/object:Api::OpAsync::Status
+    path: 'status'
+    complete: 'DONE'
+    allowed:
+      - 'PENDING'
+      - 'RUNNING'
+      - 'DONE'
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error/errors'
+    message: 'message'
+id_format: '{{project}}/{{zone}}/{{instance_group}}/{{instance}}'
+mutex: instanceGroups/{{project}}/zones/{{zone}}/{{instance_group}}
+exclude_tgc: true
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'instance_group_membership'
+    skip_test: true
+    vars:
+      network_name: 'network'
+      instance_group_name: 'instance-group'
+      instance_name: 'instance'
+custom_code: !ruby/object:Provider::Terraform::CustomCode
+  pre_delete: templates/terraform/pre_delete/compute_instance_group_membership.go.erb
+  custom_import: templates/terraform/custom_import/compute_instance_group_membership.go.erb
+  decoder: templates/terraform/decoders/instance_group_membership.go.erb
+  encoder: templates/terraform/encoders/compute_instance_group_membership.go.erb
+parameters:
+  - !ruby/object:Api::Type::ResourceRef
+    name: 'zone'
+    resource: 'Zone'
+    imports: 'name'
+    description: 'A reference to the zone where the instance group resides.'
+    url_param_only: true
+    custom_expand: 'templates/terraform/custom_expand/resourceref_with_validation.go.erb'
+  - !ruby/object:Api::Type::ResourceRef
+    name: 'instanceGroup'
+    resource: 'InstanceGroup'
+    imports: 'name'
+    description: |
+      Represents an Instance Group resource name that the instance belongs to.
+    required: true
+    url_param_only: true
+    ignore_read: true
+    diff_suppress_func: tpgresource.CompareResourceNames
+properties:
+  - !ruby/object:Api::Type::ResourceRef
+    name: 'instance'
+    description: 'An instance being added to the InstanceGroup'
+    resource: 'Instance'
+    imports: 'selfLink'
+    required: true
+    custom_expand: 'templates/terraform/custom_expand/resourceref_with_validation.go.erb'
```

Comment:
I think it would be better to add a custom flatten instead of using a custom decoder that calls the expand function. You should be able to add the following (and remove the decoder entirely):
```
custom_flatten: 'templates/terraform/custom_flatten/full_to_relative_path.erb'
```

Reply:
Done! This custom_flatten already existed so I've decided to reuse it